### PR TITLE
Add vector3 iterator

### LINKED ISSUE #33: Add Vector3::begin, Vector3::end so we can use std::sort
**Original report ([archived issue](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-math/issues/33)) by Steve Peters (Bitbucket: [Steven Peters](https://bitbucket.org/%7B2ccfed09-18b8-4921-8d58-15ef01092802%7D/), GitHub: [scpeters](https://github.com/scpeters)).**

----------------------------------------
If we add a few functions like `begin` and `end` to the Vector classes, I think we would use `std::sort`.

### PR DESCRIPTION
**Pull request by KareimGazer**
# 🎉 New feature

Closes #33 

## Summary

added iterator for **Vector3** with `begin() ` and `end()` functions.

## Test it

added test at Vector3_Test  Iterator, I could turn it to examples but I don't know how to edit the make files.

## Checklist
- [ x] Signed all commits for DCO
- [ x] Added tests
- [ ] Added example and/or tutorial
- [x ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

**Comment by KareimGazer** [comment]
@chapulina @scpeters 3 days !!?

**Comment by KareimGazer** [comment]
As a beginner open source contributor, this is disappointing.

File: src/Vector3_TEST.cc [hunk 1]
```diff
@@ -146,6 +146,22 @@ TEST(Vector3dTest, Vector3d)
   EXPECT_EQ(math::Vector3i::UnitZ, math::Vector3i(0, 0, 1));
 }
 
+/////////////////////////////////////////////////
+TEST(Vector3dTest, Iterator)
```
Comment (chapulina) [inline src/Vector3_TEST.cc:150]: Could you add some more tests to check if the iterator is robust to different situations? For example, change the value of the vector while holding an iterator and check how the values change. Compare against `end()`... Also add unit tests for all `Iterator` functions.

**Comment by chapulina** [comment]
@KareimGazer , are you planning to continue working on this PR? Let us know how we can help. Thanks!

**Comment by KareimGazer** [comment]
No, it was hard to install tools, and I didn't receive help or attention. Now, I'm busy with my engineering studies.
a better contribution and installation guide would be helpful instead of scattered wikis across versions.

**Comment by chapulina** [comment]
Ok, I'll close it for now, feel free to reopen if you ever want to get back to it.

**Pull request closed without merging**